lstchain/calib/camera/r0.py

```diff
@@ -0,0 +1,288 @@
+            Number of modules
+        """
+        roisize = 40
+        size4drs = 4096
+        n_gain = 2
+        n_pix = 7
+        for nr_clus in prange(0, n_modules):
+            for gain in prange(0, n_gain):
+                for pix in prange(0, n_pix):
+                    for k in prange(0, 4):
+                        # looking for spike A first case
+                        abspos = int(1024 - roisize - 2 - fc_old[nr_clus, gain, pix] + k * 1024 + size4drs)
+                        spike_A_position = int((abspos - fc[nr_clus, gain, pix] + size4drs) % size4drs)
+                        if (spike_A_position > 2 and spike_A_position < 38):
+                            pixel = expected_pixel_id[nr_clus*7 + pix]
+                            interpolate_spike_A(waveform, gain, spike_A_position, pixel)
+                        abspos = int(roisize - 2 + fc_old[nr_clus, gain, pix] + k * 1024 + size4drs)
+                        spike_A_position = int((abspos - fc[nr_clus, gain, pix] + size4drs) % size4drs)
+                        if (spike_A_position > 2 and spike_A_position < 38):
+                            pixel = expected_pixel_id[nr_clus*7 + pix]
+                            interpolate_spike_A(waveform, gain, spike_A_position, pixel)
+
+                    spike_b_position = int(
+                        (fc_old[nr_clus, gain, pix] - 1 - fc[nr_clus, gain, pix] + 2 * size4drs) % size4drs)
+                    if spike_b_position < roisize - 1:
+                        pixel = expected_pixel_id[nr_clus*7 + pix]
+                        interpolate_spike_B(waveform, gain, spike_b_position, pixel)
+        return waveform
+
+    def _load_calib(self):
+        """event.r0.tel[0].waveform
+        If a pedestal file has been supplied, create a array with
+        pedestal value . If it hasn't then point calibrate to
+        fake_calibrate, where nothing is done to the waveform.
+        """
+        if self.pedestal_path:
+            with fits.open(self.pedestal_path) as f:
+                n_modules = f[1].header['NAXIS4']
+                self.pedestal_value_array = np.zeros((n_modules, 2, 7, 4136), dtype=np.int16)
+                pedestal_data = np.int16(f[1].data)
+                self.pedestal_value_array[:, :, :, :self.size4drs] = pedestal_data - self.offset
+                self.pedestal_value_array[:, :, :, self.size4drs:self.size4drs + 40] \
+                    = pedestal_data[:, :, :, 0:40] - self.offset
+
+    def _get_first_capacitor(self, event, nr_module):
+        """
+        Get first capacitor values from event for nr module.
+        Parameters
+        ----------
+        event : `ctapipe` event-container
+        nr_module : number of module
+        """
+        fc = np.zeros((2, 7))
+        first_cap = event.lst.tel[0].evt.first_capacitor_id[nr_module * 8:
+                                                            (nr_module + 1) * 8]
+        for i, j in zip([0, 1, 2, 3, 4, 5, 6], [0, 0, 1, 1, 2, 2, 3]):
```

Comment:
could you add a comment on where are these arrays coming from?

Reply:
Done